src/main/java/org/sonar/plugins/github/PullRequestIssuePostJob.java

```diff
@@ -73,52 +67,38 @@ public void executeOn(Project project, SensorContext context) {
     pullRequestFacade.createOrUpdateSonarQubeStatus(report.getStatus(), report.getStatusDescription());
   }
 
-  @Override
-  public String toString() {
-    return "GitHub Pull Request Issue Publisher";
-  }
-
-  private static List<Issue> sortProjectIssues(ProjectIssues projectIssues) {
-    // Dump issues to a new list and sort it.
-    List<Issue> issues = Lists.newArrayList(projectIssues.issues());
-    Collections.sort(issues, ISSUE_COMPARATOR);
-    return issues;
-  }
-
-  private Map<InputFile, Map<Integer, StringBuilder>> processIssues(GlobalReport report) {
+  private Map<InputFile, Map<Integer, StringBuilder>> processIssues(GlobalReport report, Iterable<PostJobIssue> issues) {
     Map<InputFile, Map<Integer, StringBuilder>> commentToBeAddedByFileAndByLine = new HashMap<>();
 
-    List<Issue> issues = sortProjectIssues(projectIssues);
-    for (Issue issue : issues) {
-      if (!issue.isNew()) {
-        continue;
-      }
-      String severity = issue.severity();
-      Integer issueLine = issue.line();
-      InputFile inputFile = inputFileCache.byKey(issue.componentKey());
-      if (inputFile != null && !pullRequestFacade.hasFile(inputFile)) {
-        // SONARGITUB-13 Ignore issues on files not modified by the P/R
-        continue;
-      }
-      processIssue(report, commentToBeAddedByFileAndByLine, issue, severity, issueLine, inputFile);
-    }
+    StreamSupport.stream(issues.spliterator(), false)
+      .filter(i -> i.isNew())
+      // SONARGITUB-13 Ignore issues on files not modified by the P/R
+      .filter(i -> {
+        InputComponent inputComponent = i.inputComponent();
+        return inputComponent == null ||
+          !inputComponent.isFile() ||
+          pullRequestFacade.hasFile((InputFile) inputComponent);
+      })
+      .sorted(ISSUE_COMPARATOR)
+      .forEach(i -> {
+        processIssue(report, commentToBeAddedByFileAndByLine, i);
+      });
     return commentToBeAddedByFileAndByLine;
+
   }
 
-  private void processIssue(GlobalReport report, Map<InputFile, Map<Integer, StringBuilder>> commentToBeAddedByFileAndByLine, Issue issue, String severity,
-    @Nullable Integer issueLine,
-    @Nullable InputFile inputFile) {
+  private void processIssue(GlobalReport report, Map<InputFile, Map<Integer, StringBuilder>> commentToBeAddedByFileAndByLine, PostJobIssue issue) {
     boolean reportedInline = false;
-    if (gitHubPluginConfiguration.tryReportIssuesInline()) {
-      reportedInline = tryReportInline(commentToBeAddedByFileAndByLine, issue, severity, issueLine, inputFile);
+    if (gitHubPluginConfiguration.tryReportIssuesInline() && issue.inputComponent() != null && issue.inputComponent().isFile()) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'inputComponent' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)